File: src/main/java/org/jetbrains/fortran/lang/psi/FortranLabelDoStmt.java
package org.jetbrains.fortran.lang.psi;

import java.util.List;
import org.jetbrains.annotations.*;
import com.intellij.psi.PsiElement;

public interface FortranLabelDoStmt extends FortranBeginConstructStmt {
    @Nullable
    FortranConstructNameDecl getConstructNameDecl();

    @NotNull
    FortranLabel getLabel();

    @Nullable
    FortranLabelDecl getLabelDecl();

    @Nullable
    FortranExpr getExpr();

    @Nullable
    PsiElement getColon();

    @Nullable
    PsiElement getEol();
}

File: src/main/kotlin/org/jetbrains/fortran/lang/psi/FortranLabelDoStmt.kt
package org.jetbrains.fortran.lang.psi

import com.intellij.psi.PsiElement

interface FortranLabelDoStmt : FortranBeginConstructStmt {
    override val constructNameDecl: FortranConstructNameDecl?
    val label: FortranLabel
    val labelDecl: FortranLabelDecl?
    val expr: FortranExpr?
    val colon: PsiElement?
    val eol: PsiElement?
}